model/riscv_insts_zcmop.sail

```diff
@@ -0,0 +1,100 @@
+/*  (ERC) under the European Union’s Horizon 2020 research and innovation                */
+/*  programme (grant agreement 789108, ELVER).                                           */
+/*                                                                                       */
+/*                                                                                       */
+/*  Redistribution and use in source and binary forms, with or without                   */
+/*  modification, are permitted provided that the following conditions                   */
+/*  are met:                                                                             */
+/*  1. Redistributions of source code must retain the above copyright                    */
+/*     notice, this list of conditions and the following disclaimer.                     */
+/*  2. Redistributions in binary form must reproduce the above copyright                 */
+/*     notice, this list of conditions and the following disclaimer in                   */
+/*     the documentation and/or other materials provided with the                        */
+/*     distribution.                                                                     */
+/*                                                                                       */
+/*  THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS''                   */
+/*  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED                    */
+/*  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A                      */
+/*  PARTICULAR PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR                  */
+/*  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,                         */
+/*  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT                     */
+/*  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF                     */
+/*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND                  */
+/*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,                   */
+/*  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT                   */
+/*  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF                   */
+/*  SUCH DAMAGE.                                                                         */
+/*=======================================================================================*/
+
+/* ********************************************************************* */
+/* This file specifies the compressed insts. in the 'Zcmop' extension.   */
+
+/* These instructions are only legal if misa.C() is true and is checked in
+ * the fetch-execute logic.
+ */
+/* ****************************************************************** */
+
+union clause ast = ZCMOP : (bits(3))
+mapping clause encdec_compressed = ZCMOP(b1008) if haveZcmop()
```

Comment:
I'm not sure about the b1008 name, maybe:
```
mapping clause encdec_compressed = ZCMOP(mop) if haveZcmop()
  <-> 0b01100 @ mop : bits(3) @ 0b100000 @ 0b01
```
would be better?

Reply:
The `haveZimop` and `haveZcmop` would have to be present because a future extension that needs them would need to depend on them. Zicfiss is the first extension planning to use them and the idea is to branch out of the ZCMOP/ZIMOP for the few code points that Zicfiss uses - that would be controlled by `haveZicfiss`.

Reply:
updated to use name `mop` as the arguement.